packages/web/src/pages/sign-up-page/pages/SignUpPage.tsx

```diff
@@ -34,30 +57,148 @@ type SignUpEmailValues = {
   email: string
 }
 
+const FormSchema = z.object({
+  email: z
+    .string({ required_error: messages.invalidEmail })
+    .regex(EMAIL_REGEX, { message: messages.invalidEmail })
+})
+
 export const SignUpPage = (props: SignUpPageProps) => {
   const { onNext } = props
-
-  const handleSubmit = useCallback(
+  const dispatch = useDispatch()
+  const navigate = useNavigateToPage()
+  const [checkEmailInUse, result] = useEmailInUse()
+  const { data: emailExists, status: emailApiStatus, errorMessage } = result
+  const submitHandler = useCallback(
     (values: SignUpEmailValues) => {
       const { email } = values
-      onNext({ stage: 'create-password', params: { email } })
+      checkEmailInUse(email)
```

Comment:
we should use the formik "isSubmitting here", and use that for the loading state on submit button, happy to pair on this.

Reply:
ultimately this should be considered a "form level validation", and so lets hook it up appropriately